### DISCUSSION TITLE
TimerBot PROBLEM

### DISCUSSION THREAD

**Discussion by p17m0** [python-telegram-bot/python-telegram-bot#3439] [Q&A] [answered]
I took example and ...
AttributeError: 'NoneType' object has no attribute 'get_jobs_by_name'
How to solve it? My context.job_queue is None, why?

**Comment by Bibo-Joshi** [comment, marked as answer]
Since V20.0a5 most 3rd party dependencies are converted into optional dependencies. To use the JobQueue, install via `pip install python-telegram-bot[job-queue] --pre`. More info is available in the readme/the docs Startpage/the pypi page

**Reply by p17m0** [reply]
Thank you a lot!

**Reply by Thesoul20** [reply]
I have run the command : `pip install python-telegram-bot[job-queue] --pre` success, but the problem still appear.

**Reply by Poolitzer** [reply]
you probably run the python file with a different environment then pip installs to. How do you run it?

**Reply by vakhrushevgp** [reply]
i have the only one env -- it is in root of my mac os machine. it still didnt work after installing ptb[job-queue] --pre --break-system-packages. (the last flag is suggested by mac). 
![2025-01-29 02 18 02](https://github.com/user-attachments/assets/149a0f9c-b373-4ede-91d6-51037f58761b)
